moveit_planners/ompl/ompl_interface/src/detail/ompl_constraints.cpp

```diff
@@ -408,4 +408,106 @@ ompl::base::ConstraintPtr createOMPLConstraints(const moveit::core::RobotModelCo
   }
   return std::make_shared<ompl::base::ConstraintIntersection>(num_dofs, ompl_constraints);
 }
+
+/******************************************
+ * Toolpath Constraint
+ * ****************************************/
+ToolPathConstraint::ToolPathConstraint(const moveit::core::RobotModelConstPtr& robot_model, const std::string& group,
+                                       unsigned int num_dofs, const unsigned int num_cons, double step)
+  : BaseConstraint(robot_model, group, num_dofs, num_cons), step_(step)
+{
+  pose_nn_.setDistanceFunction([this](const EigenIsometry3dWrapper& pose1, const EigenIsometry3dWrapper& pose2) {
+    double translation_distance = (pose1.tform_.translation() - pose2.tform_.translation()).norm();
+    double rotation_distance = 0.0;  // TODO
```

Comment:
See https://github.com/ompl/ompl/blob/main/src/ompl/base/spaces/src/SO3StateSpace.cpp#L250-L277
There may be equivalent code in Eigen, tf2, ...

Reply:
It's just the arc cosine of the quaternion dot product, so easy enough to do with Eigen. Thanks!